client/src/lib.rs

```diff
@@ -288,8 +297,12 @@ pub struct ConfiguredClient<C> {
 impl<C> ConfiguredClient<C> {
     /// Set HTTP request headers overwriting previous headers
     pub fn set_headers(&self, headers: HashMap<String, String>) {
-        let mut guard = self.headers.write();
-        *guard = headers;
+        self.headers.write().user_headers = headers;
+    }
+
+    /// Set API key, overwriting previous
+    pub fn set_api_key(&self, api_key: Option<String>) {
+        self.headers.write().api_key = api_key;
     }
```

Comment:
How do you keep consistent the headers in `options.headers` and the `headers` field in ConfiguredClient? Or are we assuming `options` are just the initial frozen options?

Reply:
Options are just the initial frozen options as they always have been (but lang can make theirs mutable if they really want, which Python does but .NET doesn't)